tests/test_telemetry.py

```diff
@@ -45,7 +45,7 @@ class MockFailedResponse:
     text = "Non existent URL"
 
 
-@patch("cosmos.telemetry.httpx.get", return_value=MockFailedResponse())
+@patch.object(telemetry.httpx, "get", return_value=MockFailedResponse())
```

Comment:
Why did we introduce this and other telemetry-related changes as part of this PR?

Reply:
The new tests that I wrote were affecting these tests for Python version 3.8-3.10, although these succeeded for Python 3.11, 3.12 without this change. I guess since I was manipulating the sys.modules to remove cosmos in the new tests they were affecting here as they references were getting altered. Not sure why it works for later version of Python. I have refactored my newly added tests to undo this change and ensure they are not affected.